src/gasman.c

```diff
@@ -2082,56 +2082,43 @@ void CheckMasterPointers( void )
 }
 
 
-/****************************************************************************
-**
-*F  SwapMasterPoint( <bag1>, <bag2> ) . . . swap pointer of <bag1> and <bag2>
-*/
-void SwapMasterPoint (
-    Bag                 bag1,
-    Bag                 bag2 )
+/* swap pointer of <bag1> and <bag2> */
+void SwapMasterPoint(Bag bag1, Bag bag2)
 {
-    Bag *               ptr1;
-    Bag *               ptr2;
-    Bag swapbag;
+    Bag * ptr1;
+    Bag * ptr2;
+    Bag   swapbag;
 
-    if ( bag1 == bag2 )
+    if (bag1 == bag2)
         return;
 
-    /* get the pointers                                                    */
```

Comment:
Here is a suggested mega-comment :)

```
    // We need to make sure the correct things are GCed. There are various things
    // to consider.
    //
    // As changed bags are stored as a singly linked list, we can't easily
    // take anything out of the list of changed bags. Also, as the changed bag are
    // stored in a linked list with the pointer stored inside the bag itself,
    // when we swap PTR_BAG, we just to swap LINK_BAG as well.
    //
    // If both bags are old, if the original bag1 was changed we need to
    // now mark bag2 changed (and vice-versa).
    //
    // If both bags are new, GAP assumes they are both changed.
    //
    // If bag1 was young and bag2 was old, then after swap bag1 is old and
    // bag2 is young (the 'young'ness moves with the contents)
    //
    // Mark bag1 changed if bag2 was previously changed (for same reason as above).
    //
    // More importantly, bag2 is now young, but the only reference to bag2
    // might be in an old, unchanged bag, so it will get GCed.
    //
    // We therefore force bag2 onto the Changed bag list (we can't use CHANGED_BAG,
    // as it skips young bags). Gasman treats anything on the changed bag list as
    // a root, not garbage collecting it or anything it references.
    //
    // In practice, just always put both bags on the Changed bag list, rather than
    // pick through the exact cases (it is never incorrect to mark something
    // changed.)
```